app/controllers/application_controller.rb

```diff
@@ -2,7 +2,10 @@ class ApplicationController < ActionController::Base
   before_action :configure_permitted_parameters, if: :devise_controller?
   around_action :switch_locale
   layout :choose_layout
-  before_action :set_raven_context
+  before_action :set_raven_context, :set_locale
+  after_action :set_locale
```

Comment:
why do you need before and after action
doesn't look right
if you stored it in session it should stay there through out request